From e6cc9a0584f6cd1639b37d77059cf66a63fb1558 Mon Sep 17 00:00:00 2001
From: Yann Gautier <yann.gautier@foss.st.com>
Date: Wed, 25 Oct 2023 18:12:25 +0200
Subject: [PATCH 387/954] fix(stm32mp2): set some PM functions static

This corrects sparse warnings:
plat/st/stm32mp2/stm32mp2_pm.c:106:20: warning: symbol
 'stm32mp_pm_idle_states' was not declared. Should it be static?
plat/st/stm32mp2/stm32mp2_pm.c:120:14: warning: symbol
 'stm32mp_supported_pwr_states' was not declared. Should it be static?

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git e33d2efe2e5548eff0a2b82f311af16c7c0c7216 fix(stm32mp2): set some PM functions static]
Change-Id: I21a24fbfe2a81ab54244ee3a6aa9d565cc468144
Signed-off-by: Yann Gautier <yann.gautier@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/338364
---
 plat/st/stm32mp2/stm32mp2_pm.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/plat/st/stm32mp2/stm32mp2_pm.c b/plat/st/stm32mp2/stm32mp2_pm.c
index 8d29e61c99..68f9fba6da 100644
--- a/plat/st/stm32mp2/stm32mp2_pm.c
+++ b/plat/st/stm32mp2/stm32mp2_pm.c
@@ -129,7 +129,7 @@ u_register_t saved_scr_el3;
  *  enable us to use binary search during power state validation.
  *  The table must be terminated by a NULL entry.
  */
-const unsigned int stm32mp_pm_idle_states[] = {
+static const unsigned int stm32mp_pm_idle_states[] = {
 	PWRSTATE_RUN,
 	PWRSTATE_STOP1,
 	PWRSTATE_LP_STOP1,
@@ -143,7 +143,7 @@ const unsigned int stm32mp_pm_idle_states[] = {
 #define PM_IDLE_STATES_SIZE ARRAY_SIZE(stm32mp_pm_idle_states)
 
 /* The supported low power mode on the board, including STANDBY */
-unsigned int stm32mp_supported_pwr_states[PM_IDLE_STATES_SIZE + 1U];
+static unsigned int stm32mp_supported_pwr_states[PM_IDLE_STATES_SIZE + 1U];
 
 extern void stm32_stop2_entrypoint(void);
 
-- 
2.53.0

